Python/ki/__init__.py

```diff
@@ -1123,6 +1123,23 @@ def OnTimer(self, ID):
         elif ID == kTimers.JalakBtnDelay:
             self.SetJalakGUIButtons(1)
 
+        # Flash the scroll down arrow.
+        elif ID == kTimers.IncomingChatFlash:
+            if self.KILevel < kNormalKI:
+                mKIdialog = KIMicro.dialog
+            else:
+                mKIdialog = KIMini.dialog
+            if self.chatMgr.incomingChatFlashState > 0:
+                btn = ptGUIControlButton(mKIdialog.getControlFromTag(kGUI.miniChatScrollDown))
+                if self.chatMgr.incomingChatFlashState & 1:
+                    btn.hide()
+                else:
+                    btn.show()
+                self.chatMgr.incomingChatFlashState -= 1
+                PtAtTimeCallback(self.key, 0.15, kTimers.IncomingChatFlash)
+            else:
+                mKIdialog.refreshAllControls()
```

Comment:
Why does the entire KI need to be redrawn? This is an expensive operation according to my profiling (corresponding to several framerate stutters) from several years ago, and I find myself adverse to introducing unnecessary refreshes.

Reply:
Good point. I don’t remember. It probably doesn’t – I guess the only thing that needs to be redrawn in the proper shown or hidden state is the scroll arrow. I’ll take a look.